txtl/sbmlutil.py

```diff
@@ -0,0 +1,230 @@
+# sbmlutil.py - libsbml helper functions
+# RMM, 14 Aug 2018
+#
+# Copyright (c) 2018, Build-A-Cell. All rights reserved.
+# See LICENSE file in the project root directory for details.
+
+import libsbml
+import re
+from .parameter import Parameter
+from warnings import warn
+
+# Reaction ID number (global)
+reaction_id = 0
+
+# Create an SBML model
+def create_sbml_model():
+    document = libsbml.SBMLDocument(3, 1)
+    model = document.createModel()
+
+    # Define units for area (not used, but keeps COPASI from complaining)
+    unitdef = model.createUnitDefinition()
+    unitdef.setId('square_metre')
+    unit = unitdef.createUnit()
+    unit.setKind(libsbml.UNIT_KIND_METRE)
+    unit.setExponent(2)
+    unit.setScale(0)
+    unit.setMultiplier(1)
+
+    # Set up required units and containers
+    model.setTimeUnits('second')            # set model-wide time units
+    model.setExtentUnits('mole')            # set model units of extent
+    model.setSubstanceUnits('mole')         # set model substance units
+    model.setLengthUnits('metre')           # area units (never used?)
+    model.setAreaUnits('square_metre')      # area units (never used?)
+    model.setVolumeUnits('litre')           # default volume unit
+
+    # Define the default compartment
+    compartment = model.createCompartment()
+    compartment.setId('txtl')
+    compartment.setConstant(True)           # keep compartment size constant
+    compartment.setSpatialDimensions(3)     # 3 dimensional compartment
+    compartment.setVolume(1e-6)             # 1 microliter
+
+    return document, model, compartment
+
+# Helper function to add a species to the model
+def add_species(mixture, type, name, ic=None, debug=False):
+    model = mixture.model   # Get the model where we will store results
+    
+    # Construct the species name
+    prefix = type + " " if type != None else ""
+    species_name = prefix + name
+    
+    # Construct the species ID
+    species_id = re.sub(" ", "_", species_name)
+    species_id = re.sub("--", "_", species_id)
+    species_id = re.sub(":", "_", species_id)
+    
```

Comment:
It might be better to use [setIdFromNames](http://sbml.org/Software/libSBML/5.13.0/docs//python-api/set_id_from_names_8py-example.html) class provided as an external program by libSBML. It takes care of not just : and spaces but all other special characters and things not allowed as an SBML identifier. 

This might be helpful to use whenever we set identifier for any component (so, before all setId commands), as it will give robustness against various `name` strings that we might use.

Reply:
A simple way to use this class is -
```
allids = mixture.getAllIds()
trans = SetIdFromNames(allids)
new_id = trans.getValidIdForName(name_string)
```

The `getAllIds()` member function needs to be added (it's very simple - just iterate over `model.getListOfAllElements()` and return the list of all ids in the model).